wp-content/themes/core/components/header/subheader/Subheader_Single_Controller.php

```diff
@@ -0,0 +1,81 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Project\Templates\Components\header\subheader;
+
+use Tribe\Project\Templates\Components\link\Link_Controller;
+
+class Subheader_Single_Controller extends Subheader_Controller {
+
+	public const TAG_NAME             = 'tag_name';
+	public const TAG_LINK             = 'tag_link';
+	public const DATE                 = 'date';
+	public const AUTHOR               = 'author';
+	public const SHOULD_RENDER_BYLINE = 'should_render_byline';
+
+	private ?string $tag_name          = null;
+	private ?string $tag_link          = null;
+	private string $date               = '';
+	private string $author             = '';
+	private bool $should_render_byline = true;
+
+	public function __construct( array $args = [] ) {
+		parent::__construct( $args );
+		$args = $this->parse_args( $args );
+
+		$this->tag_name             = (string) $args[ self::TAG_NAME ];
+		$this->tag_link             = (string) $args[ self::TAG_LINK ];
+		$this->date                 = (string) $args[ self::DATE ];
+		$this->author               = (string) $args[ self::AUTHOR ];
+		$this->should_render_byline = (bool) $args[ self::SHOULD_RENDER_BYLINE ];
+	}
+
+	protected function required(): array {
+		$required = parent::required();
+
+		$required[ self::CLASSES ]         = [ 'c-subheader', 'c-subheader--single' ];
+		$required[ self::CONTENT_CLASSES ] = [ 'c-subheader__content', 'c-subheader__content--single' ];
+
+		return $required;
+	}
+
+	protected function defaults(): array {
+		$defaults = parent::defaults();
+
+		$defaults[ self::TAG_NAME ]             = null;
+		$defaults[ self::TAG_LINK ]             = null;
+		$defaults[ self::DATE ]                 = '';
+		$defaults[ self::AUTHOR ]               = '';
+		$defaults[ self::SHOULD_RENDER_BYLINE ] = true;
+
+		return $defaults;
+	}
```

Comment:
More of a question but I noticed Justin putting this info at the bottom of his PHP files recently. Do we have a standard for something like that so other Devs know where to quickly look?

Reply:
Not sure I follow, do you have an example?

Reply:
https://github.com/moderntribe/harvard-iemi/blob/develop/wp-content/themes/core/components/video/Video_Controller.php

Just that the required and defaults functions are at the bottom in the link I posted. Seems like something we could standardize where they go so if I'm moving quick I know right where to look for those

Reply:
Ah I see. IMO I put them at the top incase the controller gets really big. This is how most components are:

Const
Constructor
Defaults
Required
Everything else

Reply:
V2 of the phpcs rules will always put public functions above protected and then private so developers can properly scan the interfaces they can use.

We should start doing that now but phpcbf automatically fixes this easily.